src/coins.cpp

```diff
@@ -44,14 +44,12 @@ CCoinsMap::iterator CCoinsViewCache::FetchCoin(const COutPoint &outpoint) const
     if (it != cacheCoins.end())
         return it;
     Coin tmp;
-    if (!base->GetCoin(outpoint, tmp))
+    if (!base->GetCoin(outpoint, tmp) || tmp.IsSpent()) {
+        // If we find a spent coin in the parent cache, no
+        // need to fetch it into this cache.
```

Comment:
"no need to put it into this cache" (we already fetched it)

Reply:
The whole function we are in here is called `FetchCoin` so I would understand "fetching" as everything that is going on in his function and not the fetch into a temp variable, so for me this is ok.